staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -332,13 +332,16 @@ func (s *store) GuaranteedUpdate(
 				if err != nil {
 					return err
 				}
+				mustCheckData = false
 				if !bytes.Equal(data, origState.data) {
 					// original data changed, restart loop
-					mustCheckData = false
 					continue
 				}
 			}
-			return decode(s.codec, s.versioner, origState.data, out, origState.rev)
+			// recheck that the data from etcd is not stale before short-circuiting a write
```

Comment:
Could you please link to root issues when adding glue logic.

Reply:
you mean like a github issue? I trust the added test more than a comment

Reply:
It's more of a general request, not anything blocking. 
Folks who want to refactor code it helps having history behind the changes.  

For example "Guaranteed Update" is a throwback from the original design and as this PR shows that's a hard thing to "Guarantee".